RevenueCatUI/Data/PaywallViewConfiguration.swift

```diff
@@ -8,6 +8,7 @@
 import Foundation
 
 import RevenueCat
+import UIKit
```

Comment:
This will break for platforms that don't support `UIKit`. Let's not add this setting to `PaywallViewConfiguration` since it's only used for `PaywallViewController`.

Reply:
gotcha if that's the case then I'll have to move userInterfaceStyle out of PaywallViewConfiguration.
I'm assuming that you'd want me to create a variable userInterfaceStyle inside viewController?

Reply:
Yes exactly, since that parameter is exclusive to `PaywallViewController`.

Reply:
Cool then I'll go ahead and resolve the comments about renaming the variable in the configuration file!

Reply:
@NachoSoto 
One more question. 
So updating the configuration here actually recreates the hosting controller
   ```swift 
private var configuration: PaywallViewConfiguration {
        didSet {
            // Overriding the configuration requires re-creating the `HostingViewController`.
            // This is used by some Hybrid SDKs that require modifying the content after creation.
            self.hostingController = self.createHostingController()
        }
    }
````
if we were to exclude the userInterfaceStyle out of configuration, then it's most likely that it wouldn't rerender right? WDYT about this?

Reply:
Have you tried doing this overriding the trait in the `UIViewController`? Does that get forwarded to `SwiftUI`?

Reply:
That's an option, but better yet would be to avoid recreating it if `UIHostingController` detects the change in the `UITraitCollection`, maybe using `overrideTraitCollection(forChild childViewController: [UIViewController])`.

Reply:
@NachoSoto 
not sure if i understood you correctly, but 
without the code above,
it updates the userInterfaceStyle but does not propagate to swift UI.
with the code above,
it recreates the hosting controller which does get propagated to swiftUI

Reply:
Gotcha, I think our comments are out of sync haha. I'll try that!

Reply:
@NachoSoto 
Looks like it's working great!

https://github.com/RevenueCat/purchases-ios/assets/6516487/cbb94fe4-ac1b-417f-992a-aa080700dfa7

Added a timer to switch back and forth dark and light automatically from view controller, and it works great.